openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -99,7 +99,7 @@ class AccessTokenView(RatelimitMixin, _DispatchingView):
     def dispatch(self, request, *args, **kwargs):
         response = super(AccessTokenView, self).dispatch(request, *args, **kwargs)
 
-        token_type = request.POST.get('token_type', 'no_token_type_supplied').lower()
+        token_type = request.POST.get('token_type', request.META.get('X-Token-Type', 'no_token_type_supplied')).lower()
```

Comment:
Can you add a corresponding unit test for this change?
Here's an already existing `JWT` token_type test you can follow: https://github.com/edx/edx-platform/blob/a4f03620a41b4d8b6a0de1f3bc5581e505230b17/openedx/core/djangoapps/oauth_dispatch/tests/test_views.py#L220

See the following article on how to send an HTTP header via Django's test client: https://stackoverflow.com/questions/31902901/django-test-client-method-override-header

Reply:
Sure, is something along these lines what you had in mind? 

https://github.com/edx/edx-platform/pull/21662/files#diff-3bd6f243e7dde99b2aa4cfa1c00aee1b